miio/viomidishwasher.py

```diff
@@ -0,0 +1,529 @@
+        program = self.data["program"]
+        try:
+            return Program(program)
+        except ValueError:
+            _LOGGER.warning("Program %r is Unknown.", program)
+            return Program.Unknown
+
+    @property
+    def door_open(self) -> bool:
+        """Returns the door status of the device.
+
+        Returns:
+            Bool: True if open False if closed.
+        """
+
+        return bool(self.data["run_status"] & (1 << 7))
+
+    @property
+    def system_status_raw(self) -> int:
+        """Returns the raw status number of the device.
+
+        This is in general used to detected:
+            - Errors in the system.
+            - If the door is open or not.
+
+        Returns:
+            int: Raw value.
+        """
+
+        return self.data["run_status"]
+
+    @property
+    def status(self) -> MachineStatus:
+        """Returns the machine status of the device.
+
+        This is in general used to determine what the device is doing.
+
+        Returns:
+            MachineStatus: The machine status Enum.
+        """
+
+        return MachineStatus(self.data["wash_status"])
+
+    @property
+    def temperature(self) -> int:
+        """Returns the temperature in degree Celsius as determined by the NTC thermistor.
+
+        Returns:
+            int: degree Celsius.
+        """
+
+        return self.data["wash_temp"]
+
+    @property
+    def power(self) -> bool:
+        """Returns the power status of the device.
+
+        Returns:
+            bool: The return value. True for power on, False for power off.
+
+        Raises:
+            DeviceException: When the returned value by the device is not 1 or 0.
+        """
+
+        value = self.data["power"]
+        if value in [0, 1]:
+            return bool(value)
+
+        raise DeviceException(f"{value} is not a valid power status.")
+
+    @property
+    def time_left(self) -> timedelta:
+        """Returns the timedelta in seconds of time left of the current program.
+
+        Will always be 0 if no program is running.
+
+        Returns:
+            timedelta: A timedelta with seconds left of the current program.
+
+        Raises:
+            DeviceException: When the returned value by the device is not a valid integer.
+        """
+        value = self.data["left_time"]
+        if isinstance(value, int):
+            return timedelta(seconds=value)
+
+        raise DeviceException(f"{value} is not a valid integer for time_left.")
+
+    @property
+    def schedule(self) -> Optional[datetime]:
+        """Returns a datetime when the scheduled program should be finished..
+
+        Will always be 0 if nothing is scheduled.
+
+        Returns:
+            datetime: A datetime when the selected program should be finished.
+
+        Raises:
+            DeviceException: When the returned value by the device is not a valid integer.
+        """
+
+        value = self.data["wash_done_appointment"]
+        if isinstance(value, int):
```

Comment:
Prefer try-catch blocks over `isinstance`, that will make also writing tests much easier (isinstance does not play well with mocks).